aggsender/types/interfaces.go

```diff
@@ -159,3 +171,23 @@ type MaxL2BlockNumberLimiterInterface interface {
 	AdaptCertificate(
 		buildParams *CertificateBuildParams) (*CertificateBuildParams, error)
 }
+
+type VerifyIncomingRequest struct {
+	Certificate         *agglayertypes.Certificate
+	PreviousCertificate *agglayertypes.CertificateHeader
+}
+
+type CertificateValidator interface {
+	ValidateCertificate(ctx context.Context, params VerifyIncomingRequest) error
+}
+
+// CertificateValidateAndSigner is an interface to attach a certificate validator and signer
+// to aggsender regular flow
+type CertificateValidateAndSigner interface {
```

Comment:
We will remove this interface once we implement the actual component. Am I right?